django/utils/dateformat.py

```diff
@@ -30,11 +30,17 @@
 
 
 class Formatter(object):
+
+    def format_piece(self, piece):
+        if type(self.data) is datetime.date and hasattr(TimeFormat, piece):
+            raise TypeError('The format for date objects may not contain time-related format specifiers')
```

Comment:
Is it possible to include the invalid specifiers in the message? That would probably make it easier to debug.